pkg/config/nodetreemodel/config_test.go

```diff
@@ -25,3 +29,82 @@ func TestBuildDefaultMakesTooManyNodes(t *testing.T) {
 	_, ok = n.(LeafNode)
 	require.Equal(t, ok, true)
 }
+
+// Test that default, file, and env layers can build, get merged, and retrieve settings
+func TestBuildDefaultFileAndEnv(t *testing.T) {
+	t.Skip("fix merge to enable this test")
+
+	configData := `network_path:
+  collector:
+    workers: 6
+secret_backend_command: ./my_secret_fetcher.sh
+`
+	os.Setenv("DD_SECRET_BACKEND_TIMEOUT", "60")
+	os.Setenv("DD_NETWORK_PATH_COLLECTOR_INPUT_CHAN_SIZE", "23456")
+
+	cfg := NewConfig("test", "DD", nil)
```

Comment:
We should use something else than the official Datadog prefix, `DD_`, to avoid potential conflict with the test runner in the CI.

Reply:
Good idea, changed to `TEST_`